boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2024 Arrow Electronics.
+# SPDX-License-Identifier: Apache-2.0
+
+# General configuration
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_BUILD_OUTPUT_HEX=y
```

Comment:
@gmarull

Reply:
these do not seem board specific configs but soc level preferences. Systick may not even be needed if enabled in DT.

Reply:
I copied it from another PSoC board. I deleted `CONFIG_CORTEX_M_SYSTICK=y`, but `CONFIG_BUILD_OUTPUT_HEX=y` is required to flash an application with (Infineon's version of) openOCD.

Without this Kconfig option, you get the following error:

```shell
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner openocd
FATAL ERROR: hex file location is unknown. Try enabling CONFIG_BUILD_OUTPUT_HEX.
```

Reply:
> `CONFIG_BUILD_OUTPUT_HEX=y` is required to flash an application with (Infineon's version of) openOCD.

This means other boards are actually wrong, this soc cannot be easily flashed if there's no hex in the build folder. Add this option in the soc defconfig.

Reply:
This seems to be a more generic issue in the PSoC SoC configurations. 
I checked the other supported PSoC boards. Almost all supported PSoC boards (cyw920829m2evk_02, cy8cproto_063_ble, cy8cproto_062_4343w) cannot be flashed with Infineon's openOCD version, if you comment out this Kconfig option in the board defconfig file. For cy8ckit_062s4 only pyOCD is supported as runner and this is able to flash bin files.
I am not sure if this issue should be fixed in this pull request since this Kconfig option is missing in all PSoC SoC configurations. I think, it makes more sense to fix it in a separate pull request and then also delete it in all board defconfig files. 
I have also done a code search and found this Kconfig option in around 50 board defconfig files. So, it seems that there are several SoC configurations that do not enable this option.

Reply:
CONFIG_OUTPUT_BUILD_HEX=y is in all Cypress and Infineon PSoC6 board defconfigs; this is a requirement with our version of openOCD -- and as you've noted with many other boards as well. Is the suggestion that this should move from the board defconfig to an SOC file?

Reply:
Yes, this was the suggestion from gmarull to move this Kconfig option from the board defconfig to the SoC defconfig file. However, I think, it makes no sense to change it just for one PSoC SoC (family). If we change it, we should change it for all PSoC families and then delete it in all PSoC board configurations.
On the other side, you find this Kconfig option in a lot of board configurations. So, I am not sure if it needs be changed.
@gmarull What do you think about it?

Reply:
I think it's ok to ask to do this right for the board you're adding, other boards can be fixed later.

Reply:
I have now deleted this Kconfig option in the board defconfig file and added it to the  `soc/infineon/cat1a/psoc6_02/Kconfig.defconfig` file, as there is no separate defconfig file for each SoC. This Kconfig option is now set for all PSOC 6_02 based MCUs.